### PR TITLE
Master wowl boi

### PR DESCRIPTION
**Pull request by brboi**

### CODE REVIEW

File: addons/web/static/src/components/dialog/dialog.js [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
     useActiveElement("modal");
-    useExternalListener(window, "keydown", this._onKeydown);
+    useHotkey(
+      "escape",
+      () => {
+        if (!this.modalRef.el.classList.contains("o_inactive_modal")) {
+          this._close();
+        }
+      },
+      { altIsOptional: true }
```
Comment (aab-odoo) [inline addons/web/static/src/components/dialog/dialog.js:21]: This is a detail, but I don't like this option name. This remark is not really constructive, as I don't have any suggestion for the moment :D

File: addons/web/static/src/legacy/utils.js [hunk 2]
```diff
@@ -52,6 +52,81 @@ export function makeLegacyRpcService(legacyEnv) {
...
+            { altIsOptional: true }
+          );
+          legacyDialogs[dialogId++] = {
+            legacyDialog,
+            hotkeyToken,
+          };
+        }
+      });
+      legacyEnv.bus.on("legacy_dialog_destroyed", null, (legacyDialog) => {
+        if (legacyDialog.$modal) {
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: how can this be falsy?

File: addons/web/static/src/legacy/utils.js [hunk 3]
```diff
@@ -52,6 +52,81 @@ export function makeLegacyRpcService(legacyEnv) {
...
+          "escape",
+          () => {
+            owlDialog._close();
+          },
+          { altIsOptional: true }
+        );
+        owlDialogs[dialogId++] = {
+          owlDialog,
+          hotkeyToken,
+        };
```
Comment (aab-odoo) [inline addons/web/static/src/legacy/utils.js]: it seems that you can factorize most of the handler

File: addons/web/static/tests/legacy/core/owl_dialog_tests.js [hunk 4]
```diff
@@ -267,6 +273,58 @@ odoo.define('web.owl_dialog_tests', function (require) {
...
+
+            const { legacyEnv, env } = await makeLegacyDialogMappingTestEnv();
+
+            // OwlDialog env
+            patchWithCleanup(Dialog.prototype, {
+                setup() {
+                    this.env = legacyEnv;
+                    this._super();
+                }
+            });
```
Comment (aab-odoo) [inline addons/web/static/tests/legacy/core/owl_dialog_tests.js:288]: as far as I understand, you could get rid of this patch by more accurately spawning the legacy dialog (i.e. like we actually do in legacy): `new Dialog().open();`

We can talk about this, as I may be missing something

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/components/dialog/dialog.js (hunk 1 above)
Comment (brboi) [reply addons/web/static/src/components/dialog/dialog.js:21]: As ALT is in fact CONTROL on MacOS, maybe we should find a name that is not directly related to a key name ?

Maybe something like renaming the whole "ALT" concept to a more generic name: `hyper` or `gui` or `leader` ...

Or maybe just renaming this option name to a generic one: `{ safe: false }` (=== `{ altIsOptional: true }`)

What do you think ?

File: addons/web/static/src/legacy/utils.js (hunk 2 above)
Comment (brboi) [reply addons/web/static/src/legacy/utils.js]:
```
const unopenedModal = new LegacyDialog(null, {});
unopenedModal.close();
```

File: addons/web/static/src/components/dialog/dialog.js (hunk 1 above)
Comment (brboi) [reply addons/web/static/src/components/dialog/dialog.js:21]: `{ modRequired: false }`

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**